Interface-manager part 1

daniel-noland opened the pull request:
Part one of the long-awaited interface manager pr.

I expect a part two on top of this shortly.

Fredi-raspall commented on net/src/interface/vtep.rs:
```diff
@@ -0,0 +1,57 @@
+    /// The TTL of the vtep.
+    /// This value can be `None` if the use did not specify the value when the VTEP was created or,
+    /// for whatever reason, the value is not available in the netlink message describing the vtep.
+    #[builder(default = Some(0))]
```
Could it be that Mac has been forgotten here?

Fredi-raspall replied:
I'd make it optional, if included.

daniel-noland replied:
Mac is part of the parent structure

Fredi-raspall commented on net/src/ipv4/addr.rs:
```diff
@@ -6,6 +6,39 @@
+    ///
+    /// Returns the supplied address back in the [`Err`] case if it is not a unicast address.
+    pub fn new(ip: Ipv4Addr) -> Result<SourceIpv4Addr, Ipv4Addr> {
+        if ip.is_multicast() || ip.is_unspecified() {
```
In theory the unspecified ip address 0.0.0.0 can appear in some packets (e.g. DHCP).
I don't know if forbidding them here may cause trouble depending on the use of `SourceIpv4Addr` ?

daniel-noland replied:
That is a really good point.

I know that linux  forbids 0.0.0.0 as the source address of a vtep

Fredi-raspall reviewed (changes requested):
@daniel-noland looks good to me. The only comments I have relate to SourceIpv4Addr and the MAC for the vxlan interface.

daniel-noland commented:
@Fredi-raspall

I addressed the thing about SourceIpv4.  That was a very good point.  That type was not sound.

I also added one commit at the end to clarify how mac is handled.

qmonnet commented on net/src/route/mod.rs:
```diff
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Open Network Fabric Authors
+
+//! Common abstractions used in routing.
```
Discussed during the meeting - I understand we're fine with this for now, and we'll try to refactor it One Day (TM). So I'm OK with the `route` submodule.

qmonnet reviewed (approved):
Looks good to me, thanks!
(I'd recommend squashing the last two commits into the relevant prior commits before merging.)
